libs/nio/src/main/java/org/elasticsearch/nio/Page.java

```diff
@@ -61,16 +61,21 @@ public void close() {
 
     private static class RefCountedCloseable extends AbstractRefCounted {
 
-        private final Runnable closeable;
+        private final Closeable closeable;
```

Comment:
I think `Releasable` would be enough here, we never throw anything let alone an `IOException`? We'd need to move `Releasable` into `:libs:elasticsearch-core` to be accessible here but that's ok IMO.

Reply:
++ let's move `Releasable` :)